tests/kinematics/DiffDriveKinematicsTest.cpp

```diff
@@ -60,4 +54,53 @@ TEST_CASE("Differential Drive Kinematics Test")
 	// drive CW in a full circle with radius 1
 	assertApprox({0, 0, 0},
 				 kinematics.getNextPose(wheelVel(3 * PI / 4, PI / 4), {0, 0, 0}, 4));
-}
\ No newline at end of file
+
+	constexpr double max_wheel_vel = 0.75;
+
+	// Proportional Preservation Tests
+	assertApprox({0.5, 0, 0.5},
+				 kinematics.ensureWithinWheelSpeedLimit(
+					 kinematics::DiffDriveKinematics::PreferredVelPreservation::Proportional,
```

Comment:
We have `using namespace kinematics` at the top, so you can simplify this to `DiffDriveKimatics::...`, both here and everywhere else further down.